Emby.Server.Implementations/Metrics/PrometheusMetricsCollector.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Entities.Audio;
+using MediaBrowser.Controller.Entities.Movies;
+using MediaBrowser.Controller.Entities.TV;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Metrics;
+using Prometheus;
+using Prometheus.DotNetRuntime;
+
+namespace Emby.Server.Implementations.Metrics
+{
+    /// <summary>
+    /// Prometheus backend for metrics.
+    /// </summary>
+    public class PrometheusMetricsCollector : IMetricsCollector
+    {
+        private readonly ILibraryManager _libraryManager;
+
+        private static readonly Gauge _itemCountGauge = Prometheus.Metrics.CreateGauge("jellyfin_library_items_total", "The number of items in the library", new[] { "item_type" });
+        private static readonly string[] _metricTypes = new string[]
+        {
+            nameof(Movie),
+            nameof(Series), nameof(Season), nameof(Episode),
+            nameof(MusicArtist), nameof(MusicAlbum), nameof(MusicVideo), nameof(Audio),
+            nameof(Book),
+            nameof(PhotoAlbum), nameof(Photo)
+        };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrometheusMetricsCollector" /> class.
+        /// </summary>
+        /// <param name="libraryManager">The library manager</param>
+        public PrometheusMetricsCollector(ILibraryManager libraryManager)
+        {
+            _libraryManager = libraryManager;
+            _libraryManager.ItemAdded += IncrementItemCount;
+            _libraryManager.ItemRemoved += (s, e) => UpdateItemCount();
```

Comment:
`ItemRemoved` only seems to provide the parent item for e.g. If a series is removed, it fires once for the series but not the episodes. Calling `GetRecursiveChildren()` on it does not give them either. Is there another way to get all individual items removed?